doc/ssdmond/ssdmond-hld.md

```diff
@@ -0,0 +1,129 @@
+# SONiC SSD Daemon Design #
+### Rev 0.1 ###
+
+ | Rev |     Date    |       Author       | Change Description                |
+ |:---:|:-----------:|:------------------:|-----------------------------------|
+ | 0.1 |             |      Ashwin Srinivasan  | Initial version                   |
+
+## 1. Overview
+
+This document is intended to provide a high-level design for a Solid-State Drive monitoring daemon.
+
+Solid-State Drives (SSDs) are storage devices that use NAND-flash technology to store data. They offer the end user significant benefits compared to HDDs, some of which include reliability, reduced size, increased energy efficiency and improved IO speeds which translates to faster boot times, quicker computational capabilities and an improved system responsiveness overall. Like all devices, however, they experience performance degradation over time on account of a variety of factors such as overall disk writes, bad-blocks management, lack of free space, sub-optimal operational temperature and good-old wear-and-tear which speaks to the overall health of the SSD. 
+
+The goal of the SSD Monitoring Daemon (ssdmond) is to provide meaningful metrics for the aforementioned issues and enable streaming telemetry for these attributes so that the required preventative measures are triggered in the eventuality of performance degradation.
+
+## 2. Data Collection
+
+We are intrested in the following characteristics that describe various aspects of the SSD:
+
+### **2.1 Priority 0 Attributes** 
+
+**The following are dynamic fields, offering up-to-date information that describes the current state of the SSD**
+
+    - IO Reads
```

Comment:
@assrinivasan Are these count number of bytes written by the filesystem or the number of writes written to the disk which accounts the write amplification.

Reply:
These are the number of bytes written by the filesystem. I will research/modify this section to refer to bytes read from and written to disk.

Reply:
Addressed in latest commit @prgeor